### PR TITLE
expression: implement vectorized evaluation for 'builtinBitOrSig' 

### PR DESCRIPTION
**Pull request by snithish**
### What problem does this PR solve? 

PR implements vectorized  `builtinBitOrSig` from #12103 

### What is changed and how it works?

`vecEvalInt` of `builtinBitOrSig` has been implemented and `vectorized()` has been enables

```
BenchmarkVectorizedBuiltinOpFunc/builtinBitOrSig-VecBuiltinFunc-8              	  160938	      7328 ns/op
BenchmarkVectorizedBuiltinOpFunc/builtinBitOrSig-NonVecBuiltinFunc-8           	   23313	     51289 ns/op
```
### Check List 

Tests 

 - Unit test

Code changes

 - Has enables vectorized execution for `BitOr`

### CODE REVIEW

File: expression/builtin_op_vec.go [hunk 1]
```diff
@@ -91,11 +91,42 @@ func (b *builtinLogicOrSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column)
...
+
+		isNull := false
+
+		if !isNull0 && !isNull1 {
+			arg0s[i] = arg0s[i] | arg1s[i]
+		} else {
+			arg0s[i] = 0
+			isNull = true
+		}
+		result.SetNull(i, isNull)
```
Comment (SunRunAway) [inline expression/builtin_op_vec.go]: we can use [`Column.MergeNulls`](https://godoc.org/github.com/pingcap/tidb/util/chunk#Column.MergeNulls) to calculate nulls before the for loop. Then use `result.IsNull(i)` to skip the null result in for loop.

Comment (snithish) [reply expression/builtin_op_vec.go]: `Column.MergeNulls` was quite helpful, I have made this change and pushed in a commit

File: expression/builtin_op_vec.go [hunk 2]
```diff
@@ -91,11 +91,36 @@ func (b *builtinLogicOrSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column)
...
+	arg0s := result.Int64s()
+	arg1s := buf.Int64s()
+
+	result.MergeNulls(buf)
+
+	for i := 0; i < numRows; i++ {
+		if result.IsNull(i) {
+			continue
+		}
+		arg0s[i] = arg0s[i] | arg1s[i]
```
Comment (SunRunAway) [inline expression/builtin_op_vec.go]: We can use arg0s[i] |= arg1s[i] for the same code style.

File: expression/builtin_op_vec.go [hunk 3]
```diff
@@ -91,11 +91,36 @@ func (b *builtinLogicOrSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column)
...
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[1].VecEvalInt(b.ctx, input, buf); err != nil {
+		return err
+	}
+
+	arg0s := result.Int64s()
+	arg1s := buf.Int64s()
+
```
Comment (XuHuaiyu) [inline expression/builtin_op_vec.go]: useless empty lines 114 and 116

File: expression/builtin_op_vec.go [hunk 4]
```diff
@@ -91,11 +91,36 @@ func (b *builtinLogicOrSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column)
...
+	result.MergeNulls(buf)
+
+	for i := 0; i < numRows; i++ {
+		if result.IsNull(i) {
+			continue
+		}
```
Comment (XuHuaiyu) [inline expression/builtin_op_vec.go]: We can remove this check here, nothings changes but be more friendly to the CPU branch prediction.

File: expression/builtin_op_vec.go (hunk 3 above)
Comment (snithish) [reply expression/builtin_op_vec.go]: Pushed in a new commit to remove redundant empty lines

File: expression/builtin_op_vec.go (hunk 4 above)
Comment (snithish) [reply expression/builtin_op_vec.go]: That makes a lot of sense, pushed in a commit to incorporate this

**Pull request merged**